### DISCUSSION TITLE
Nous A1Z offers update, but firmware date and version are empty

### DISCUSSION THREAD

**Discussion by steff75** [Koenkk/zigbee2mqtt#14208] [Devices]
Hello,

I have a new Nous A1Z plug connected to Zigbee2MQTT. This now offers me an OTA update in the web interface. However, one can not see from which version to which version the update should take place.

![Screenshot 2022-09-27 093811](https://user-images.githubusercontent.com/19283026/192468464-c8c509fc-0077-47f9-86fb-ad60da43ae7c.png)

Since I have read that some firmwares cause problems with poweroffs with the Tuya TS011F sockets, I am not sure if I should do this update. The [discussion ](https://github.com/Koenkk/zigbee2mqtt/discussions/13343)also says that there should already be a v3.0 firmware, but so far it seems that you can only update it with the Tuya Gateway.

I haven't connected anything to the plug yet. Should I do the update or wait and see? 

Thanks!

**Comment by stefanchiriac** [comment]
the power off might be one of the issues. On me is reporting wrong power even not in use. I did the OTA update from zigbee2mqtt and nothing changed.

**Comment by steff75** [comment]
Ok, thanks. I just want to prevent, that I install a worse version with the update.

**Comment by Jordan87** [comment]
I think its a common issue not only for TS011F that you dont have info about firmware version in Z2M.
![image](https://user-images.githubusercontent.com/10414918/193288503-1da9c3b7-4fca-4f36-9d13-df4c3430b0c2.png)

But here you have a hint how to check it manually ;)
Go to a device>Dev console>Cluster "genBasic" and attribute "appVersion" and press "Read"
![image](https://user-images.githubusercontent.com/10414918/193289046-eb276140-fb8d-43c1-8297-d6825a561643.png)
In my case for this specific device it gives value 77
Decimal 77 = Binary 01001101
![image](https://user-images.githubusercontent.com/10414918/193289591-45b665eb-108b-4baa-86ee-49049a7ab62c.png)
They are splited in to three parts 01-00-1101
binary 01 = decimal 1
binary 00 = decimal 0
binary 1101 = decimal 13
So now you know that current fw version is 1.0.13
For installed 3.0.0 (latest Nous) it will show decimal 192 ;)

**Reply by steff75** [reply]
@Jordan87 Thank you very much for the tutorial! :)

**Comment by polygon242** [comment]
the problem is still present. no version info on the device / OTA page but the update infos state the fw version.
![nous az1 ota](https://github.com/Koenkk/zigbee2mqtt/assets/119975123/fb75615b-f2a8-4c59-a0f1-544ba7a2d707)

**Comment by hpuac** [comment]
Hey folks, I can confirm that, the firmware update works (see last nous device), but it's not showing any version in the UI:
<img width="1238" alt="Screenshot 2024-01-29 at 10 41 55" src="https://github.com/Koenkk/zigbee2mqtt/assets/1850100/33fb5f42-2a19-4b00-b918-1cedf5924e44">

**Reply by cracyfloyd** [reply]
Same here !!

**Comment by jerem2772** [comment]
I have the same issue

**Comment by projectfreak** [comment]
Same here... :-(

**Comment by hulkito-nol** [comment]
Same on my side...

**Comment by evilmumi** [comment]
still same in 1.39.1
